src/corehost/cli/apphost/bundle/bundle_runner.cpp

```diff
@@ -166,28 +167,9 @@ void bundle_runner_t::reopen_host_for_reading()
     }
 }
 
-// Checks if this host binary has a valid bundle signature.
-// If so, it sets header_offset and returns true.
-// If not, returns false.
-bool bundle_runner_t::process_manifest_footer(int64_t &header_offset)
+void bundle_runner_t::process_manifest_header()
 {
-    seek(m_bundle_stream, -manifest_footer_t::num_bytes_read(), SEEK_END);
-
-    manifest_footer_t* footer = manifest_footer_t::read(m_bundle_stream);
-
-    if (!footer->is_valid())
-    {
-        trace::info(_X("This executable is not recognized as a .net core bundle."));
-        return false;
-    }
-
-    header_offset = footer->manifest_header_offset();
-    return true;
-}
-
-void bundle_runner_t::process_manifest_header(int64_t header_offset)
-{
-    seek(m_bundle_stream, header_offset, SEEK_SET);
+    seek(m_bundle_stream, marker_t::header_offset(), SEEK_SET);
 
     manifest_header_t* header = manifest_header_t::read(m_bundle_stream);
```

Comment:
Since `header` is not stored anywhere, and it's allocated in `manifest_header_t`, the memory leaks.  It might be a better idea to allocate on the stack here and have `manifest_header_t::read()` take a point to it as a parameter.

Reply:
@lpereira  please see this commit
https://github.com/dotnet/core-setup/pull/6885/commits/bd597daecedfba1995832d66d720b37d35b08c39